### PR TITLE
[HDInsight] Add ListBillingSpecs API to HDInsight's  stable and preview version

### PR DESCRIPTION
**Pull request by aim-for-better**
Add ListBillingSpecs API  to HDInsight's two api version(stable/2018-06-01-preview, preview/2015-03-01-preview)
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (nschonni) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]:
```suggestion
      "description": "This class represent a single filter object that defines a multidimensional set. The dimensions of this set are Regions, ClusterFlavors, NodeTypes and ClusterVersions. The dimensions of this set are Regions, ClusterFlavors, NodeTypes and ClusterVersions. The constraint should be defined based on the following: FilterMode (Exclude vs Include), VMSizes (the VM sizes in affect of exclusion/inclusion) and the ordering of the Filters. Later filters override previous settings if conflicted.",
```

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (nschonni) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]:
```suggestion
      "description": "This class represent a single filter object that defines a multidimensional set. The dimensions of this set are Regions, ClusterFlavors, NodeTypes and ClusterVersions. The dimensions of this set are Regions, ClusterFlavors, NodeTypes and ClusterVersions. The constraint should be defined based on the following: FilterMode (Exclude vs Include), VMSizes (the VM sizes in affect of exclusion/inclusion) and the ordering of the Filters. Later filters override previous settings if conflicted.",
```

**Review by nschonni — COMMENTED** [review]
exlucsion -> exclusion

**Comment by idear1203** [comment]
LGTM.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: > Thanks for your suggestion. I have updated the description.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: > 

Thanks for your suggestion. I have updated the description.

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (sergey-shandar) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: Why is there `preview` version inside `stable` folder?

Comment (aim-for-better) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: > Why is there `preview` version inside `stable` folder?

@idear1203  could you please help me reply this?

Comment (idear1203) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: @sergey-shandar This is intentional. It’s a little confusing, but per rule [#5](https://github.com/Azure/azure-rest-api-specs#directory-structure) here, whether the swagger spec is in the preview or stable folder “is not a direct analog for whether or not an API Version has the ‘-preview’ suffix or not. SDKs that are generated from 'preview' folder items should indicate to their customers in the most idiomatic way that breaking changes may still be coming.”
 
Basically the short version is that HDInsight is a GA service that still has an API with a -preview tag. Somebody made the decision several years ago to release stable versions of Hydra/Hyak SDKs for .NET, JavaScript, etc. that consumed this the -preview API. We are now releasing stable swagger-based SDKs so we can at least finally deprecate the Hydra/Hyak SDKs. As a result, we are making a commitment that we will not make breaking changes to this swagger spec version so we can GA some swagger-based SDKs.

**Comment by idear1203** [comment]
@yungezz Is there a ETA for ARM feedback? We have been waiting for several days

**Comment by idear1203** [comment]
@NelsonDaniel Could we get some ARM feedback? This PR is blocking for one week now

**Comment by NelsonDaniel** [comment]
@idear1203 I am not the ARM reviewer. As stated on top of the PR: "If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html)."

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: The type field must be added in every model that does not have it. In this case, type: object must be added. For your reference: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#understanding-the-importance-of-type-keyword-while-defining-model-types

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: From the description there is a limited set of values this will accept. Would it be better to have an x-ms-enum?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: From the description there is a limited set of values this will accept. Would it be better to have an x-ms-enum?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: type: object

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: type: object

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: from the description there is just two states. Would bool or enum fit better here?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: from the description looks like you have a limited set of values. Would an x-ms-enum be a better fit?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: type: object

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: type: object

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: type: object

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: Would something other than string make more sense since you have only two states?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: type: object

**Review by NelsonDaniel — CHANGES_REQUESTED** [review]

**Comment by idear1203** [comment]
@NelsonDaniel Thank you for the information! We have notified ARM team for a review. Meanwhile, we are updating code per your comments:)

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (idear1203) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: There seems other values. Let's keep it as it is now

**Comment by idear1203** [comment]
Hi @NelsonDaniel , after adding type: object, we got the following model validation error. Could you please help take a look? 

```
operationId: Locations_ListBillingSpecs |  
-- | --
  | scenario: Get the subscription billingSpecs for the specified location |  
  | source: response |  
  | responseCode: '200' |  
  | severity: 0 |  
  | code: INVALID_TYPE |  
  | details: |  
  | code: INVALID_TYPE |  
  | params: |  
  | - object |  
  | - undefined |  
  | message: Expected type object but found type undefined |  
  | path: '' |  
  | title: '#/definitions/BillingResponseListResult' |  
  | description: >- |  
  | The response for the operation to get regional billingSpecs for a |  
  | subscription. |  
  | position: |  
  | line: 166 |  
  | column: 34 |  
  | url: >- |  
  | specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json |  
  | directives: {} |  
  | jsonPath: ''
```

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Comment by idear1203** [comment]
> Hi @NelsonDaniel , after adding type: object, we got the following model validation error. Could you please help take a look?
> 
> ```
> 
> Validating "examples" and "x-ms-examples" in  specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json: |  
> -- | --
>   |   |  
>   | error : |  
>   | operationId: Locations_ListBillingSpecs |  
>   | scenario: Get the subscription billingSpecs for the specified location |  
>   | source: response |  
>   | responseCode: '200' |  
>   | severity: 0 |  
>   | code: INVALID_TYPE |  
>   | details: |  
>   | code: INVALID_TYPE |  
>   | params: |  
>   | - object |  
>   | - undefined |  
>   | message: Expected type object but found type undefined |  
>   | path: '' |  
>   | title: '#/definitions/BillingResponseListResult' |  
>   | description: >- |  
>   | The response for the operation to get regional billingSpecs for a |  
>   | subscription. |  
>   | position: |  
>   | line: 381 |  
>   | column: 34 |  
>   | url: >- |  
>   | specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json |  
>   | directives: {} |  
>   | jsonPath: '' |  
>   |   |  
>   | Validating "examples" and "x-ms-examples" in  specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json: |  
>   |   |  
>   | error : |  
>   | operationId: Locations_ListBillingSpecs |  
>   | scenario: Get the subscription billingSpecs for the specified location |  
>   | source: response |  
>   | responseCode: '200' |  
>   | severity: 0 |  
>   | code: INVALID_TYPE |  
>   | details: |  
>   | code: INVALID_TYPE |  
>   | params: |  
>   | - object |  
>   | - undefined |  
>   | message: Expected type object but found type undefined |  
>   | path: '' |  
>   | title: '#/definitions/BillingResponseListResult' |  
>   | description: >- |  
>   | The response for the operation to get regional billingSpecs for a |  
>   | subscription. |  
>   | position: |  
>   | line: 166 |  
>   | column: 34 |  
>   | url: >- |  
>   | specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json |  
>   | directives: {} |  
>   | jsonPath: '' |  
>   |   |  
>   | ##[error]Bash exited with code '1'.
> ```

Resolved by correct example file: change "Body" to "body"

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: I think this should be "modelAsString" : true
@ravbhatnagar, do you agree?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (NelsonDaniel) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: I think this should be "modelAsString" : true
@ravbhatnagar, do you agree?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (NelsonDaniel) [inline specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: I think this should be "modelAsString" : true
@ravbhatnagar, do you agree?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (NelsonDaniel) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: I think this should be "modelAsString" : true
@ravbhatnagar, do you agree?

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (idear1203) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: We want to reuse the exising Tier definition [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/cluster.json#L910-L921). If we change "modelAsString" as true for this `Tier` definition, we will get the following error:
```
FATAL: System.InvalidOperationException: Swagger document contains two or more x-ms-enum extensions with the same name 'Tier' and different values: Standard,Premium vs. Standard,Premium
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/ObjectBuilder.cs:line 150
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 217
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 48
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/ObjectBuilder.cs:line 182
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 217
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 48
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/ObjectBuilder.cs:line 182
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 217
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 48
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.SwaggerModeler.BuildCompositeTypes() in /opt/vsts/work/1/s/src/SwaggerModeler.cs:line 348
   at AutoRest.Modeler.SwaggerModeler.Build(ServiceDefinition serviceDefinition) in /opt/vsts/work/1/s/src/SwaggerModeler.cs:line 66
   at AutoRest.Modeler.Program.<ProcessInternal>d__2.MoveNext() in /opt/vsts/work/1/s/src/Program.cs:line 60
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NewPlugin.<Process>d__15.MoveNext()
FATAL: csharp/imodeler1 - FAILED
FATAL: Error: Plugin imodeler1 reported failure.
Process() cancelled due to exception : Plugin imodeler1 reported failure.
  Error: Plugin imodeler1 reported failure.
```

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (idear1203) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: We want to reuse the exising OSType definition [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/cluster.json#L898-L909). If we change "modelAsString" as true for this `OSType` definition, we will get the following error:
```
FATAL: System.InvalidOperationException: Swagger document contains two or more x-ms-enum extensions with the same name 'OSType' and different values: Windows,Linux vs. Windows,Linux
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/ObjectBuilder.cs:line 150
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 217
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 48
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/ObjectBuilder.cs:line 182
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 217
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 48
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/ObjectBuilder.cs:line 182
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 217
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 48
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 133
   at AutoRest.Modeler.SwaggerModeler.BuildCompositeTypes() in /opt/vsts/work/1/s/src/SwaggerModeler.cs:line 348
   at AutoRest.Modeler.SwaggerModeler.Build(ServiceDefinition serviceDefinition) in /opt/vsts/work/1/s/src/SwaggerModeler.cs:line 66
   at AutoRest.Modeler.Program.<ProcessInternal>d__2.MoveNext() in /opt/vsts/work/1/s/src/Program.cs:line 60
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NewPlugin.<Process>d__15.MoveNext()
FATAL: csharp/imodeler1 - FAILED
FATAL: Error: Plugin imodeler1 reported failure.
Process() cancelled due to exception : Plugin imodeler1 reported failure.
  Error: Plugin imodeler1 reported failure.
```

Comment (NelsonDaniel) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: That definition should also be corrected

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (NelsonDaniel) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: That definition should also be corrected

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (ravbhatnagar) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: yes sure

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (ravbhatnagar) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: sure

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (idear1203) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: We cannot do this because our SDK is GA now and it will be a breaking change. We will fix the issue in the next API version. Created a GitHub issue #6505  to track this.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json
Comment (idear1203) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json]: We cannot do this because our SDK is GA now and it will be a breaking change. We will fix the issue in the next API version. Created a GitHub issue #6505  to track this.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (idear1203) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: We cannot do this because our SDK is GA now and it will be a breaking change. We will fix the issue in the next API version. Created a GitHub issue #6505  to track this.

File: specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json
Comment (idear1203) [reply specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/locations.json]: We cannot do this because our SDK is GA now and it will be a breaking change. We will fix the issue in the next API version. Created a GitHub issue #6505  to track this.

**Review by NelsonDaniel — APPROVED** [review]
In the future x-ms-enums should "modelAsString": true to avoid breaking changes in the generated types. @idear1203 created an issue here https://github.com/Azure/azure-rest-api-specs/issues/6505

**Pull request merged**
